### PR TITLE
Keep the transient google communication errors in the poll queue Closes #1665

### LINKED ISSUE #1665: The JES polling actor unexpectedly terminated while conducting polls. Making a new one...
I was running Cromwell with a JES backend.  I got through about 300 out of 1000 workflows that were submitted as a batch; the rest are still listed as `Running`.  Oddly, the genomics API operations report that they are still running though there are no instances still up.  I am not sure if this is a Cromwell issue or a Google one, but here is what I am seeing from Cromwell.   Cromwell has been working well for smaller batches.  Any thoughts? 

Thanks!

```
[ERROR] [11/12/2016 02:22:12.170] [cromwell-system-akka.actor.default-dispatcher-2686] [akka://cromwell-system/user/cromwell-service/$b/$a] The JES polling actor Actor[akka://cromwell-system/user/cromwell-service/$b/$a/$8#-1275882726] unexpectedly terminated while conducting 11 polls. Making a new one...
[INFO] [11/12/2016 02:22:12.170] [cromwell-system-akka.actor.default-dispatcher-2686] [akka://cromwell-system/user/cromwell-service/$b/$a] watching Actor[akka://cromwell-system/user/cromwell-service/$b/$a/$9#-1852863496]
[WARN] [11/12/2016 02:22:12.171] [cromwell-system-akka.dispatchers.backend-dispatcher-2573] [akka://cromwell-system/user/cromwell-service/WorkflowManagerActor/WorkflowActor-7f0d7c16-d434-4dd7-a7ba-c7e897701c9a/WorkflowExecutionActor-7f0d7c16-d434-4dd7-a7ba-c7e897701c9a/7f0d7c16-d434-4dd7-a7ba-c7e897701c9a-EngineJobExecutionActor-salmonRun.salmonQuant:NA:1/7f0d7c16-d434-4dd7-a7ba-c7e897701c9a-BackendJobExecutionActor-7f0d7c16:salmonRun.salmonQuant:-1:1/JesAsyncBackendJobExecutionActor] JesAsyncBackendJobExecutionActor [UUID(7f0d7c16)salmonRun.salmonQuant:NA:1]: Caught exception, retrying:
java.lang.RuntimeException: Unexpected actor death!
	at cromwell.backend.impl.jes.statuspolling.JesPollingActorClient$$anonfun$pollingActorClientReceive$1.applyOrElse(JesPollingActorClient.scala:33)
	at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:170)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
	at cromwell.backend.impl.jes.JesAsyncBackendJobExecutionActor.aroundReceive(JesAsyncBackendJobExecutionActor.scala:82)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:526)
	at akka.actor.ActorCell.invoke(ActorCell.scala:495)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

File: supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/statuspolling/JesApiQueryManagerSpec.scala [hunk 1]
```diff
@@ -97,10 +97,6 @@ class JesApiQueryManagerSpec extends TestKitSuite("JesApiQueryManagerSpec") with
         val probe = statusRequesters(index)
         probe.expectNoMsg(max = AwaitAlmostNothing)
       }
-      (batchOffset until batchOffset + BatchSize) foreach { index =>
-        val probe = statusRequesters(index)
-        probe.expectMsg(max = TestExecutionTimeout, hint = s"Polling error to requester #$index", obj = JesPollingActor.JesPollError)
-      }
```
Comment (cjllanwarne) [inline supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/statuspolling/JesApiQueryManagerSpec.scala]: We should also check the queue is appropriately refreshed. Maybe as part of the eventually on line 110? If I'm reading the test correctly:
```
eventually { 
  ...
  jaqmActor.underlyingActor.workQueue.size should be(2 * BatchSize - batchOffset) }
```
Also while you're here I'm pretty sure we need to fix some lines:
- `[90] ... sender = statusPoller2`
- `[93] stopMethod(statusPoller2)`

**Comment by cjllanwarne** [comment]
👍

**Comment by geoffjentry** [comment]
@cjllanwarne re retry: i went back and forth on that topic and managed to come up with an argument I felt was valid the other way. This definition ended up being far simpler so I chose the short path.

**Comment by cjllanwarne** [comment]
On hold while Jeff is out

**Comment by kshakir** [comment]
LGTM, and happy to go over in person later if we'd like. 👍

**Comment by mcovarr** [comment]
Not sure why @kshakir's thumb is not honored by pullapprove, let's see if it likes mine any better.  :+1:

**Pull request merged**